samples/apps/autogen-studio/autogenstudio/utils/utils.py

```diff
@@ -259,6 +261,58 @@ def init_app_folders(app_file_path: str) -> Dict[str, str]:
     return folders
 
 
+def parse_skill_for_docs(module_code: str, title: str = "") -> str:
+    """
+    Analyzes the source code of a Python module to extract information about top-level functions,
+    including the name, docstring, parameters with their types and default values.
+
+    :param module_code: str, the source code of the Python module.
+    :param title: str, the name of the skill (function) we want.
+    :return: json dumps of the list of dicts,
+            a list containing dictionaries with information about each top-level function.
```

Comment:
Can you add an example of output. Its seems like a json string but the structure isn't clear.

Reply:
1st example:
[{"function_name": "generate_and_save_images", "docstring": "Function to paint, draw or illustrate images based on the users query or request. Generates images from a given query using OpenAI's DALL-E model and saves them to disk.  Use the code below anytime there is a request to create an image.\n\n:param query: A natural language description of the image to be generated.\n:param image_size: The size of the image to be generated. (default is \"1024x1024\")\n:return: A list of filenames for the saved images.", "parameters": [{"name": "query", "type": "str", "default": null}, {"name": "image_size", "type": "str", "default": "'1024x1024'"}]}]

2nd:
[{"function_name": "save_cat_ascii_art_to_png", "docstring": "Creates ASCII art of a cat and saves it to a PNG file.\n\n:param filename: str, the name of the PNG file to save the ASCII art.", "parameters": [{"name": "filename", "type": null, "default": "'ascii_cat.png'"}]}]